src/main/java/uk/gov/cshr/service/InviteService.java

```diff
@@ -46,17 +47,29 @@ public Invite findByCode(String code) {
         return inviteRepository.findByCode(code);
     }
 
+    @ReadOnlyProperty
+    public Invite findByForEmail(String email) {
+        return inviteRepository.findByForEmail(email);
+    }
+
+    @ReadOnlyProperty
+    public Optional<Invite> findByForEmailAndStatus(String email, InviteStatus status) {
+        return inviteRepository.findByForEmailAndStatus(email, status);
+    }
+
     public boolean isCodeExpired(String code) {
         Invite invite = inviteRepository.findByCode(code);
+        return isInviteCodeExpired(invite);
+    }
+
+    public boolean isInviteCodeExpired(Invite invite) {
         long diffInMs = new Date().getTime() - invite.getInvitedAt().getTime();
+        boolean codeExpired = false;
```

Comment:
This variable can be removed. Please see below two comments.

Reply:
Done.